File: analytics-sdk/analytics-storage-server/src/controllers/databaseController.js
const {
  createDatabaseAndTable,
  insertData,
  updateData,
  deleteTable,
} = require('../services/databaseService');

async function handleCreate(req, res, next) {
  try {
    const { dbName, tableName } = req.body;
    await createDatabaseAndTable(dbName, tableName);
    res.status(201).json({ success: true, message: 'Database and table ready' });
  } catch (err) {
    next(err);
  }
}

async function handleInsert(req, res, next) {
  try {
    const { dbName, tableName, data } = req.body;
    await insertData(dbName, tableName, data);
    res.status(201).json({ success: true, message: 'Data inserted' });
  } catch (err) {
    next(err);
  }
}

async function handleUpdate(req, res, next) {
  try {
    const { dbName, tableName, data } = req.body;
    await updateData(dbName, tableName, data);
    res.status(200).json({ success: true, message: 'Data updated' });
  } catch (err) {
    next(err);
  }
}

async function handleDelete(req, res, next) {
  try {
    const { dbName, tableName } = req.body;
    await deleteTable(dbName, tableName);
    res.status(200).json({ success: true, message: 'Table deleted' });
  } catch (err) {
    next(err);
  }
}

module.exports = {
  handleCreate,
  handleInsert,
  handleUpdate,
  handleDelete,
};

File: analytics-sdk/analytics-storage-server/src/controllers/database_controller.py
from fastapi import Request
from src.services.database_service import create_database_and_table, insert_data, update_data, delete_data

async def handle_create(request: Request):
    body = await request.json()
    await create_database_and_table(body['dbName'], body['tableName'])
    return {"success": True, "message": "Database and table ready"}

async def handle_insert(request: Request):
    body = await request.json()
    await insert_data(body['dbName'], body['tableName'], body['data'])
    return {"success": True, "message": "Data inserted"}

async def handle_update(request: Request):
    body = await request.json()
    await update_data(body['dbName'], body['tableName'], body['data'])
    return {"success": True, "message": "Data updated"}

async def handle_delete(request: Request):
    body = await request.json()
    await delete_data(body['dbName'], body['tableName'])
    return {"success": True, "message": "Table deleted"}